[F11-3] Gamified Task Manager

jeffkwoh opened the pull request.

jelneo commented on docs/DeveloperGuide.adoc:
```diff
@@ -1,961 +1,483 @@
+- Should be able to hold up to 1000 tasks without a noticable decline in performance.
+- Should not take more than 3 seconds to complete a command.
+- System should be maintainable with proper documentation.
+- Proper feedback to user when error occurs. // @jelena: Usability
```
Remember to remove my name :)

jelneo commented on docs/DeveloperGuide.adoc:
```diff
@@ -1,961 +1,483 @@
+#### MSS
+* 1. User fails to complete tasks on time
+* 2. Application displays points lost and current current points to user.
+### Extensions
```
I think there should be another `#` here

jelneo reviewed (commented):
Good effort, continue working towards v1.1.

jelneo commented:
Good effort in achieving v1.1 and please resolve the merge conflict.

Some general feedback:
- Travis build status badge should point to the build status of your own repo, not se-edu/AB-4. Other badges should be updated to match your project or removed
- Update this:
![image](https://user-images.githubusercontent.com/28497440/46580222-5d3a5380-ca53-11e8-92d0-b3b017ededc5.png)
-  I think there is a problem with rendering the user guide and developer guide on netlify, please check it out

Feedback on About Us page:
- Please upload a higher quality image for Jeffrey
- Should assign the different components in the app to members

Feedback on User Guide:
- Include all the sections that the AB4 User Guide has. Refer to the AB4 User guide as an example. Remember to update the relevant portions in the guide to fit the application that you are developing
- You can include all the features/commands that you have brainstormed and then add ‘coming in v2.0’ for features that are not implemented yet.

Feedback on Developer Guide:
- Appendices should be at the bottom of the dev guide
- Spelling error:
![image](https://user-images.githubusercontent.com/28497440/46580362-e9e61100-ca55-11e8-97de-4eb4f1731c9d.png)

pengnam commented:
@unsatisfiedpopcorn @jiaxiu11 @remo5000 @chikchengyao 
I hope this tags us all.

jelneo commented:
Good job on meeting milestone v1.2 @unsatisfiedpopcorn @pengnam @jiaxiu11 @remo5000 @chikchengyao 

Some general feedback:
- Please upload a better quality photo of Jeffrey in the About Us page. He looks really blurry here:
![image](https://user-images.githubusercontent.com/28497440/47264943-ae1d7200-d552-11e8-89a0-5ea1d4a61df7.png)
- User and dev guide not rendered properly on netlify

Some feedback on developer guide:
- developer guide is still not rendered properly on netlify
- the order of contents should follow AB-4 Dev guide. E.g. 
![image](https://user-images.githubusercontent.com/28497440/47265077-e7ef7800-d554-11e8-8e0b-b859e3148f2f.png) with appendices at the bottom
- 3.1 Achievement mechanism
![image](https://user-images.githubusercontent.com/28497440/47265151-fc804000-d555-11e8-8189-a24c6fcc59d1.png)
   1) Double-check if the circled relationships (AchievementRecord and ModelManager) are supposed to be an aggregation (I'm not saying they are wrong. It depends on how you design and implement it). It is mentioned that `AchievementRecord` 

>  stores the achievement information of the user internally with an Xp field and a Level field. When the task manager application runs for the first time, it is initialized at 0 xp and level 1, xp and level information in `AchievementRecord` are updated subsequently when tasks are completed.

Hence, I understood it as each `AchievementRecord` will always have `Xp` and `Level`.
    2) Rearrange and enlarge the `Level` box so that 'enumeration' does not spread across two lines
   
-------------------------------------------------------------------------------------------------------------
![image](https://user-images.githubusercontent.com/28497440/47265281-2175b280-d558-11e8-8e7f-12f18cd3ceed.png)


in red: arrange the elements in the diagram such that the words don't overlap and the text don't spill over to the next line
in blue: move the lifeline to the center of the box
optional box: the bars should be in front of the box and the 'opt' and condition shouldnot overlap with the activation bars
 
  -------------------------------------------------------------------------------------------------------------
![image](https://user-images.githubusercontent.com/28497440/47265316-c1334080-d558-11e8-8522-f134f07b09d0.png)


in red: same comments as above about text formatting
The Ui and EventsCenter can switch positions so that it will be clearer that some external class is calling a method in EventsCenter

   -------------------------------------------------------------------------------------------------------------

![image](https://user-images.githubusercontent.com/28497440/47265331-2129e700-d559-11e8-912e-38d082e68dd8.png)


in red: same comments as above about text formatting
It would be better if the flow of the process is from top to bottom instead of left to right.
   -------------------------------------------------------------------------------------------------------------
- 3.2 Batch Completion via Label feature
    1) Incorrect usage of `::`: `Label::many` . see this link: https://www.codementor.io/eh3rrera/using-java-8-method-reference-du10866vx
    2) It would be good to label your diagrams and then refer to it in the steps in the usage scenario
   -------------------------------------------------------------------------------------------------------------
![image](https://user-images.githubusercontent.com/28497440/47265657-0017c500-d55e-11e8-84bd-efd946a3065d.png)


correct grammar in picture above

   -------------------------------------------------------------------------------------------------------------
![image](https://user-images.githubusercontent.com/28497440/47265736-1c683180-d55f-11e8-9dd8-8549716b1f85.png)
It would be good to rephrase the part pointed to by the arrow to make the explanation clearer.
Also, include diagrams that are taught in the module such as sequence diagrams
- 3.3 Model Rollback feature
Include diagrams that are taught in the module such as sequence diagrams
- 3.4 Dependency Feature Modelling
Possible to elaborate more using the sequence diagram shift it into the 'Current implementation' section.
Sequence diagram can be rearranged and spaced out such that text and other image elements don't overlap. Please save the diagram as an image before using it to remove the squiggly red lines under some words.
Align the arrows properly such that they touch and overshoot the activation bars.

jeffkwoh commented:
@jelneo I tried not to have a high-resolution image of myself as much as I can. Apologies, I should have clarified this earlier, I pixelated the image according to the guidelines found in the cs2103 website:
> If you are uncomfortable posting your photo due to security reasons, you can post a lower resolution image...

I can replace with a more acceptable pixelated image if possible. I hope this is okay

jelneo commented:
Some things to note:
- Make sure you have contextualize the dev guide according to your product.
- for the image below, is the underlined portion a typo?
![image](https://user-images.githubusercontent.com/28497440/48263616-aff79880-e461-11e8-8dc0-6dc191e719ac.png)
- to be safe, add a return arrow for self-invocation:
![image](https://user-images.githubusercontent.com/28497440/48263846-6fe4e580-e462-11e8-8fa0-39dde2fd97d9.png)
- is there a repeated word here?
![image](https://user-images.githubusercontent.com/28497440/48263901-9c98fd00-e462-11e8-9fa2-9fee9e03f115.png)
- for the image below:
1. make it into one line
2. & 5. the arrows overshoot the activation bars
3. missing arrowhead
4. create an outline for the stacked activation box. include a return arrow for it too.
![image](https://user-images.githubusercontent.com/28497440/48264184-9ce5c800-e463-11e8-92ed-367f860f9ed0.png)
- for the image below, ensure that the two ends of the arrow touches the activation bars (don't leave nay gaps). 
1. expand the box and shift model to the right so that the diagram does not look squished.
2. the arrowhead should be right at the start of the activation bar.
![image](https://user-images.githubusercontent.com/28497440/48264427-6d838b00-e464-11e8-89b5-f2b6788fccf5.png)
- create an outline for the stacked activation box and spread them out a bit vertically so it doesn't look cluttered:
![image](https://user-images.githubusercontent.com/28497440/48264473-91df6780-e464-11e8-8bd2-f83d07dc4184.png)
- Lastly, take a look at my previous comments regarding your dev guide if you have not done so.

chikchengyao committed:
Fix priority value overflow bug

pengnam committed:
Fix overflow bug in comparator

pengnam committed:
Add dependency from completed to uncompleted check

pengnam committed:
Fix function signature

pengnam committed:
Add cehckTaskDependencyAtIndexes check at dependencycommand

chikchengyao committed:
Fix integer overflows in PriorityMode

chikchengyao committed:
Limit priority value to 9999

jiaxiu11 committed:
Add guards for integer overflow

chikchengyao committed:
Fix failing test cases

pengnam committed:
Refactor task name in DependencyCommand

remo5000 committed:
Add getDependencyCount method

remo5000 committed:
TaskCard: Add UI for dependencyCount

remo5000 committed:
Add dependencyCount to TaskCardHandle

jiaxiu11 committed:
Fix grammatical error in UserGuide

pengnam committed:
Add notes to DependencyCommand

pengnam committed:
Add cyclic dependency diagram

pengnam committed:
Merge pull request #257 from pengnam/docs

Bug fixing

remo5000 committed:
Truncate label names

1) Added a static function that truncates the label name
2) Used this function in TaskCard and TaskViewPanel

chikchengyao committed:
Fix failing tests

pengnam committed:
Amend variable renaming

chikchengyao committed:
Merge pull request #258 from jiaxiu11/int-overflow

Add guards for integer overflow to AchievementRecord

jiaxiu11 committed:
Merge pull request #254 from chikchengyao/fix-integer-overflow

Fix priority value overflow bug

chikchengyao committed:
Add newline to end of config.json

pengnam committed:
Add concatDependency, spliceDependency to Task

resolves problem wrt law of demeter in creating new tasks in DependencyCommand

pengnam committed:
Amend method in UniqueTaskList

remo5000 committed:
Fix positioning of windows

pengnam committed:
Create updateHash method in Task, refactor other methods

remo5000 committed:
Fix effective due date label being small

jiaxiu11 committed:
Fix UserGuide and DeveloperGuide formatting problems

remo5000 committed:
IsNotBlockedPredicate: Add isStatusCompleted check

pengnam committed:
Refactor getDependency to getDependencies

remo5000 committed:
Fix padding

jiaxiu11 committed:
Merge pull request #259 from pengnam/docs

Refactor non-mutable task update operations

jiaxiu11 committed:
Update achievements DeveloperGuide

jiaxiu11 committed:
Modify comments in AchievementRecord

jeffkwoh committed:
Replace "MAGIC NUMBER"

chikchengyao committed:
Show mode details upon calling ModeCommand

chikchengyao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into persist-game-mode

jeffkwoh committed:
CompleteIndex:Minor style fix for readability

pengnam committed:
Add Manual Testing section for dependency

jiaxiu11 committed:
Add check for max int exceeded in ach storage

jiaxiu11 committed:
Add max int constrains to Xp class

jeffkwoh committed:
Refactor CompleteLabelCommand

- Improve Javadocs
- Remove redundant use of iterator

remo5000 committed:
Fix style errors

remo5000 committed:
Fix style errors

pengnam committed:
Merge pull request #264 from remo5000/dependency-count

Add dependency count to TaskCard

jeffkwoh committed:
Refactor CompleteCommandParser

Apply SLAP
Update Javedocs

jeffkwoh committed:
Fix bug

Previously, no ParseExceptions are thrown if a label is invalid.

Now, using parseLabels(...) as a guard, it will throw an exception

jeffkwoh committed:
CompleteCommandParser: Update comment

Make the usage of a guard more explicit.

chikchengyao committed:
Fix some style checks

pengnam committed:
Add topological order to manual testing

pengnam committed:
Add topoorder in manual testing

jeffkwoh committed:
Change LabelMatchesKeywordPredicate behaviour

- Now named LabelMatchesAnyKeywordPredicate

- A set of keylabels can be supplied to the predicate. On a Task's label
match with any of them, it returns true

jeffkwoh committed:
Predicate: Refactor keywords to keyLabels

jeffkwoh committed:
Update tests

jeffkwoh committed:
Refactor CompleteCommandParser

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

pengnam committed:
Refactor dependency count

jeffkwoh committed:
Refactor CompleteLabelCommandTest

chikchengyao committed:
Fix failing style checks

jeffkwoh committed:
Remove extra blank lines

remo5000 committed:
Update failing test

remo5000 committed:
Merge pull request #265 from jiaxiu11/refactoring

Refactoring achievement codes

pengnam committed:
Add newline at eof for dev guide

jeffkwoh committed:
Update LabelsBuilder

- Add Javadocs
- Add non null guard

remo5000 committed:
Merge branch 'master' into truncate-fields

jeffkwoh committed:
Improve javadocs for TypicalTasks

chikchengyao committed:
Fix failing style checks

jeffkwoh committed:
Remove extra blank line

remo5000 committed:
Fix another failing test

remo5000 committed:
Merge pull request #263 from remo5000/truncate-fields

Truncate Task labels

remo5000 committed:
Change spacing of fields

remo5000 committed:
Uncomment tests

pengnam committed:
Format file (missing new lines, spaces)

pengnam committed:
Readd sections in DevGuide

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

# Conflicts:
#	src/main/java/seedu/address/ui/TaskCard.java

pengnam committed:
Merge pull request #247 from unsatisfiedpopcorn/improve-code-quality

Complete Command Bug Fixes and Improvements to code quality

jeffkwoh committed:
Add Regression Tests for Parser

remo5000 committed:
Change end of day/week/month to the acutal end

remo5000 committed:
ListCommandTest: Re-add tests

jeffkwoh committed:
Add back History MSS

This is not obseolete.

remo5000 committed:
Fix style error

remo5000 committed:
Update ListCommandSequenceDiagram

pengnam committed:
Amend javadocs

remo5000 committed:
Fix diagram

remo5000 committed:
Update image

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

# Conflicts:
#	src/test/java/seedu/address/logic/commands/CompleteLabelCommandTest.java

pengnam committed:
Remove unecessary comment

pengnam committed:
Fix cyclic check test

pengnam committed:
Improve dependency docs

pengnam committed:
Merge pull request #271 from remo5000/list-docs-update

Update ListCommand diagram

remo5000 committed:
Add fields section to UserGuide

pengnam committed:
Merge pull request #262 from remo5000/fix-ui-layout

Fix UI Layout

jiaxiu11 committed:
Merge pull request #270 from remo5000/refactor-list-time-tests

Refactor ListCommand time-based tests

chikchengyao committed:
Refactor 18 -> 19

chikchengyao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into refactor-year

jiaxiu11 committed:
Merge pull request #269 from unsatisfiedpopcorn/update-dev-guide

Add back History MSS

jiaxiu11 committed:
Merge pull request #261 from remo5000/list-doable-exclude-completed

ListCommand: Exclude completed tasks from "doable" filter

jiaxiu11 committed:
Add ach manual testing to developer guide

jiaxiu11 committed:
Fix typp

jiaxiu11 committed:
Merge branch 'master' into ach-manual-testing

jiaxiu11 committed:
Add new line to EOF

jiaxiu11 committed:
Merge pull request #268 from unsatisfiedpopcorn/improve-code-quality

Add Regression Tests for CompleteCommandParser

jiaxiu11 committed:
Merge pull request #273 from chikchengyao/refactor-year

Refactor year from 2018 to 2019

pengnam committed:
Improve documentation for dependency in devguide

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

# Conflicts:
#	docs/DeveloperGuide.adoc
#	src/main/java/seedu/address/model/task/IsNotBlockedPredicate.java

remo5000 committed:
Remove extra whitespace

remo5000 committed:
Remove TAG naming

jiaxiu11 committed:
Change Icon

remo5000 committed:
Fix statuses

jiaxiu11 committed:
Merge pull request #266 from pengnam/docs

Update docs + mild refactoring

jiaxiu11 committed:
Merge branch 'master' into ach-manual-testing

chikchengyao committed:
Fix failing test compilation

remo5000 committed:
remo5000 PPP: Add Minor feature

chikchengyao committed:
Update Ui.png

pengnam committed:
Add effective due date diagrams

pengnam committed:
Add Effective Due Date use case

remo5000 committed:
Merge remote-tracking branch 'origin/fields-docs' into ppp

remo5000 committed:
Add task-fields section to UserGuide contributions

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

pengnam committed:
Change dependency message

pengnam committed:
Resolve dependency bug

pengnam committed:
Amend comment in UniqueTaskList

pengnam committed:
Merge branch 'master' into fields-docs

pengnam committed:
Improve delete command message

pengnam committed:
Update PPP

pengnam committed:
Merge pull request #272 from CS2103-AY1819S1-F11-3/fields-docs

UserGuide: Add documentation for editable and derived fields

chikchengyao committed:
Update User Guide

remo5000 committed:
Merge branch 'master' into ppp

Commit:
Merge pull request #275 from jiaxiu11/icon

Change App Icon

jeffkwoh committed:
Update RollbackModelSequenceDiagram

jeffkwoh committed:
Add Complete Command Activity Diagram

jeffkwoh committed:
Update Complete Command MSS

Ensures that all use cases terminates

jiaxiu11 committed:
Modify phrasing

jiaxiu11 committed:
Fix documentation formatting

jiaxiu11 committed:
Modify contact us info

chikchengyao committed:
Merge pull request #280 from jiaxiu11/UG

Remove se-edu header bar

chikchengyao committed:
Fix trailing whitespace

pengnam committed:
Amend docs

pengnam committed:
Amend delete command message

pengnam committed:
Fix delete bug

pengnam committed:
Fix variable name

Commit:
Merge pull request #276 from remo5000/ppp

Vignesh PPP: Add minor feature

Commit:
Merge pull request #274 from jiaxiu11/ach-manual-testing

Add dev guide manual testing for achievement mechanism

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

# Conflicts:
#	docs/DeveloperGuide.adoc

pengnam committed:
Merge pull request #277 from pengnam/docs

Update docs, fix bug

jeffkwoh committed:
Update complete tasks use cases

jeffkwoh committed:
Merge branch 'master' into update-dev-guide

jeffkwoh committed:
Add tags to use case complete

chikchengyao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into persist-game-mode

jeffkwoh committed:
Refactor for LabelMatchesAnyKewordsPredicate

jeffkwoh committed:
Update docs

- Change image size
- Remove undo/redo statement in complete command
- Remove redundant appendix prepended header

jeffkwoh committed:
Add Manual Testing for Complete Command

jeffkwoh committed:
Add commit reference for complete command

- I.e. Documentation is written with reference to details on that commit

chikchengyao committed:
Rewrite ModeCommand to remove LoD violation

chikchengyao committed:
Fix stubs not implementing new method

chikchengyao committed:
Add logger to XmlAdaptedGameManager

jeffkwoh committed:
Update Complete via Label implementation

pengnam committed:
Add dependency command activity diagram

pengnam committed:
Amend docs

pengnam committed:
Update ShowTopologicalOrder sequence diagram

Commit:
Merge pull request #279 from unsatisfiedpopcorn/update-dev-guide

Update Dev Guide for Complete Command

chikchengyao committed:
Add diagrams

chikchengyao committed:
Add images

pengnam committed:
Update DeveloperGuide

pengnam committed:
Update developer guide

chikchengyao committed:
Update diagrams

chikchengyao committed:
Update Sequence Diagram c.f. Jelena

jeffkwoh committed:
Update adoc showing multiple label complete

chikchengyao committed:
Update Developer Guide

Commit:
Merge pull request #281 from unsatisfiedpopcorn/update-user-guide

Update User Guide to illustrate complete via multiple labels

pengnam committed:
Update docs

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

pengnam committed:
Update DeveloperGuide

pengnam committed:
Update effective due date

jeffkwoh committed:
Update PPP Overview and Minor Enhancements

chikchengyao committed:
Edit Developer Guide

chikchengyao committed:
Fix whitespace issue in GameManagerClassDiagram

chikchengyao committed:
Merge team master

chikchengyao committed:
Merge pull request #253 from chikchengyao/persist-game-mode

Persist game mode

chikchengyao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into game-documentation

pengnam committed:
Amend docs

jeffkwoh committed:
Update dev guide, add preamble

remo5000 committed:
Add list filter manual tests

remo5000 committed:
Space out stuff in ppp

jeffkwoh committed:
Update dev guide

jiaxiu11 committed:
Modify dev guide and ppp

chikchengyao committed:
Fix gamemodes awarding negative xp

jeffkwoh committed:
Update PPP

jeffkwoh committed:
Merge branch 'master' into jeff-prepare-docs-for-submission

remo5000 committed:
Add task view panel manual testing

jeffkwoh committed:
Update dev guide phrasing for label implmentation

remo5000 committed:
Add note for fields

chikchengyao committed:
Handle missing or invalid xml values for game manager

jeffkwoh committed:
Explicate CompleteCommand's dependency on GameMode

remo5000 committed:
Priority -> Priority Value

remo5000 committed:
Add Dependencies field

pengnam committed:
Add effective due date implementation in dev guide

chikchengyao committed:
Fix trailing whitespace

pengnam committed:
Add effective due date implementation

jeffkwoh committed:
Update CompleteCommand Test comments

chikchengyao committed:
Merge pull request #283 from remo5000/manual-testing

Add Manual Testing section for ListCommand filters and TaskViewPanel UI

pengnam committed:
Amend position of diagrams in EDD

pengnam committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-F11-3/main into docs

chikchengyao committed:
Merge vig's doc

chikchengyao committed:
Update UI image again

pengnam committed:
Update PPP

jeffkwoh committed:
Add Trivial Implementation of checking for Level Up

remo5000 committed:
Merge pull request #282 from unsatisfiedpopcorn/jeff-prepare-docs-for-submission

Updates PPP, Dev Guide and User Guide for Jeff's portions

jiaxiu11 committed:
Modify Manual testing

remo5000 committed:
Fix PPP titel

remo5000 committed:
Change Target User Profile

chikchengyao committed:
Fix IncreasingMode

remo5000 committed:
Add a word

remo5000 committed:
Add tags for PPP

jeffkwoh committed:
Update Diagram

pengnam committed:
Update docs

jeffkwoh committed:
Fix user guide rendering issue

pengnam committed:
Update diagrams

pengnam committed:
Merge pull request #285 from remo5000/user-profile

Add Target User Profile

chikchengyao committed:
Update docs

remo5000 committed:
Merge pull request #284 from pengnam/docs

Docs for 1.4

pengnam committed:
Add logger to UniqueTaskList

jiaxiu11 committed:
Add time diagram in user guide

jeffkwoh committed:
Final update to PPP

jeffkwoh committed:
Prune obseolete design portion of documenetation

jeffkwoh committed:
Revert "Add Trivial Implementation of checking for Level Up"

This reverts commit 8fbf96e60c9effb41d01da57a2580e3251a47d7b.

chikchengyao committed:
Merge from master

jeffkwoh committed:
Add TODO to CompleteLabelCommandTest comment

jiaxiu11 committed:
Merge pull request #287 from jiaxiu11/1.4-doc

Achievement 1.4 doc

pengnam committed:
Edit ppp

chikchengyao committed:
Fix minor labelling issue in Dev Guide use cases

remo5000 committed:
Update README

Commit:
Merge pull request #288 from unsatisfiedpopcorn/update-complete-command-test

Update to comments of complete command tests

Commit:
Merge pull request #286 from unsatisfiedpopcorn/jeff-prepare-docs-for-submission

Update guides for submission

pengnam committed:
Amend docs

jeffkwoh committed:
Add commit reference to dev guide section

Commit:
Merge pull request #291 from unsatisfiedpopcorn/add-commit-reference-to-section

Add commit reference to dev guide section

pengnam committed:
Merge pull request #289 from pengnam/docs

Add logger to UniqueTaskList

jiaxiu11 committed:
Merge pull request #290 from remo5000/update-readme

Update README

remo5000 committed:
Merge pull request #278 from chikchengyao/game-documentation

Update Ui.png and edit GameMode documentation

pengnam committed:
Update summary

jiaxiu11 committed:
Merge pull request #292 from pengnam/docs

Update summary

pengnam committed:
Add newline mss

pengnam committed:
Remove undoredo

chikchengyao committed:
Edit docs

jeffkwoh committed:
Minor Grammatical fix

remo5000 committed:
Merge pull request #295 from pengnam/docs

Docs

remo5000 committed:
Merge pull request #294 from unsatisfiedpopcorn/fix-grammar

Minor Grammatical fix

remo5000 committed:
Merge pull request #293 from chikchengyao/final-docs-update

Edit docs
